File: YK_CANSDK_CPP/examples/basic_moves.cpp
/* Low-speed movement examples. Run only after the commissioning checklist.
 * C++ port of examples/basic_moves.py.
 */

#include <iostream>

#include "ykcan/vehicle.hpp"

int main() {
    try {
        yk_can::VehicleConfig config = yk_can::VehicleConfig::from_json("config.example.json");
        yk_can::FourWheelVehicle car(config);
        car.connect();
        car.forward(0.15, 1.0);
        car.turn_left(0.15, 0.08, 1.0);
        car.turn_right(0.15, 0.08, 1.0);
        car.backward(0.15, 1.0);
        car.spin_left(0.12, 0.5);
        car.stop();
        car.close();
        return 0;
    } catch (const std::exception& exc) {
        std::cerr << "error: " << exc.what() << "\n";
        return 1;
    }
}

File: YK_CANSDK/examples/basic_moves.py
"""Low-speed movement examples. Run only after the commissioning checklist."""

from yk_can_sdk import FourWheelVehicle, VehicleConfig


def main() -> None:
    config = VehicleConfig.from_json("config.example.json")
    with FourWheelVehicle(config) as car:
        car.forward(speed=0.15, duration_s=1.0)
        car.turn_left(speed=0.15, turn=0.08, duration_s=1.0)
        car.turn_right(speed=0.15, turn=0.08, duration_s=1.0)
        car.backward(speed=0.15, duration_s=1.0)
        car.spin_left(speed=0.12, duration_s=0.5)
        car.stop()


if __name__ == "__main__":
    main()

